include/zim/archive.h

```diff
@@ -462,6 +462,20 @@ namespace zim
        */
       bool hasNewNamespaceScheme() const;
 
+      /** Get if the zim archive may have alias entries.
+       *
+       * Alias entries are entries (two at least) pointing to the same
+       * content (same cluster/blob id).
+       *
+       * This method is just a hint.
+       * If true, its means that zim archive has been created with a implementation
+       * allowing alias creation.
+       * If false, it means that zim archive has been created before we
+       * formalize alias concept. But it would be still valid (from the spec)
+       * to have alias in the archive.
+       */
+      bool mayHaveAliasEntry() const;
```

Comment:
The potential benefits of having such a method are not clear. If the implementation supporting alias creation records (e.g. in metadata) the information about actual presence of aliases in a ZIM archive (and that is made part of the ZIM spec) then this method could be replaced with a tri-value (`true`, `false`, `unknown`)  `hasAliases()` which is more meaningful.

Reply:
> The potential benefits of having such a method are not clear.

Just a hint which can be used at reader level (for example) to change behaviour. I believe we will use this just to perform checks in a differentiated manner based of ZIM version.

> If the implementation supporting alias creation records (e.g. in metadata) the information about actual presence of aliases in a ZIM archive (and that is made part of the ZIM spec) then this method could be replaced with a tri-value (`true`, `false`, `unknown`)  `hasAliases()` which is more meaningful.

There is no real quick way to know AFAIK that we have aliases and therefore don't think this is really doable to do that efficiently. The "may" versuin seems good enough.

Reply:
I want to add that if this new method just generate to much lack of consensus. On my side, this is OK to remove it (which means implicitly that for know we always consider a ZIM having - potentially - aliases).

Reply:
@veloman-yunkan Sorry I have merged a bit quickly... I agree with your last comment, but the core of the discussion as been already run at https://github.com/openzim/libzim/pull/833#issuecomment-1794900510?! I believe it is good like this. If it is proven we need more, we will do more.